### PR TITLE
repo/refs: Convert to new code style

### PR DESCRIPTION
**Pull request by cgwalters**
I happened to be reading this one recently for a reason I forget,
and it's a relatively easy conversion.

Also one not conflicting with any outstanding patches.

### CODE REVIEW

**Comment by cgwalters** [comment]
From one of the Travis runs:
```
../src/libostree/ostree-sepolicy.c:31:37: fatal error: ostree-sepolicy-private.h: No such file or directory

 #include "ostree-sepolicy-private.h"
```

Weird 😕.  It's clearly there in 7b2370dc869ce5ba76769853ef7fd0b081b15010 and lots of other things would break without it.  And `srcdir != builddir` works.  Dunno.

File: src/libostree/ostree-repo-refs.c [hunk 1]
```diff
@@ -839,51 +767,42 @@ _ostree_repo_write_ref (OstreeRepo    *self,
...
         }
 
       dfd = glnx_opendirat_with_errno (refs_remotes_dfd, remote, TRUE);
       if (dfd < 0 && (errno != ENOENT || rev != NULL))
-        {
-          glnx_set_error_from_errno (error);
-          g_prefix_error (error, "Opening remotes/ dir %s: ", remote);
-          goto out;
-        }
+        return glnx_throw_errno_prefix (error, "Opening remotes/ dir %s: ", remote);
```
Comment (jlebon) [inline src/libostree/ostree-repo-refs.c]: The glnx version already appends the `: `, so we can drop it here.

Comment (cgwalters) [inline src/libostree/ostree-repo-refs.c]: Well spotted!  I took this opportunity to learn a little bit more about the `make syntax-check` stuff we've been carrying around for a while.  Fix below ⬇️

**Comment by cgwalters** [comment]
```
$ make syntax-check
...
src/libostree/ostree-repo-refs.c:782:        return glnx_throw_errno_prefix (error, "Opening remotes/ dir %s: ", remote);
maint.mk: don't add trailing : for glnx_throw_errno_prefix
cfg.mk:27: recipe for target 'sc_glnx_errno_prefix_colon' failed
make: *** [sc_glnx_errno_prefix_colon] Error 1
make: *** Waiting for unfinished jobs....
```

**Comment by jlebon** [comment]
@rh-atomic-bot e7c8f6b

**Comment by cgwalters** [comment]
@jlebon I think you need to add a character [there](https://github.com/ostreedev/ostree/pull/752#issuecomment-288820174) (so to speak).

**Comment by jlebon** [comment]
Whoops! @rh-atomic-bot r+ e7c8f6b

**Pull request closed without merging**